app/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertExecutionRepository.java

```diff
@@ -0,0 +1,194 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseAlertExecutionRepository.class);
+
+    private final AtomicBoolean _isOpen = new AtomicBoolean(false);
+    private final EbeanServer _ebeanServer;
+    private final DatabaseExecutionHelper<AlertEvaluationResult, AlertExecution> _helper;
+
+    /**
+     * Public constructor.
+     *
+     * @param ebeanServer Play's {@code EbeanServer} for this repository.
+     */
+    @Inject
+    public DatabaseAlertExecutionRepository(@Named("metrics_portal") final EbeanServer ebeanServer) {
+        _ebeanServer = ebeanServer;
+        _helper = new DatabaseExecutionHelper<>(LOGGER, _ebeanServer, this::findOrCreateAlertExecution);
+
+    }
+
+    private AlertExecution findOrCreateAlertExecution(
+            final UUID jobId,
+            final Organization organization,
+            final Instant scheduled
+    ) {
+        final Optional<models.ebean.Organization> org = models.ebean.Organization.findByOrganization(_ebeanServer, organization);
+        if (!org.isPresent()) {
+            final String message = String.format(
+                    "Could not find org with organization.uuid=%s",
+                    organization.getId()
+            );
+            throw new EntityNotFoundException(message);
+        }
+        final Optional<AlertExecution> existingExecution = org.flatMap(r ->
+                _ebeanServer.createQuery(AlertExecution.class)
+                        .where()
+                        .eq("organization", org.get())
+                        .eq("scheduled", scheduled)
+                        .findOneOrEmpty()
+        );
+        final AlertExecution newOrUpdatedExecution = existingExecution.orElse(new AlertExecution());
+        newOrUpdatedExecution.setAlertId(jobId);
+        newOrUpdatedExecution.setOrganization(org.get());
+        newOrUpdatedExecution.setScheduled(scheduled);
+        return newOrUpdatedExecution;
+    }
+
+    @Override
+    public void open() {
+        assertIsOpen(false);
+        LOGGER.debug().setMessage("Opening DatabaseAlertExecutionRepository").log();
+        _isOpen.set(true);
+    }
+
+    @Override
+    public void close() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Closing DatabaseAlertExecutionRepository").log();
+        _isOpen.set(false);
+    }
+
+    @Override
+    public Optional<JobExecution<AlertEvaluationResult>> getLastScheduled(final UUID jobId, final Organization organization)
+            throws NoSuchElementException {
+        assertIsOpen();
+        return _ebeanServer.find(AlertExecution.class)
+                .where()
+                .eq("alert_id", jobId)
+                .eq("organization.uuid", organization.getId())
+                .setMaxRows(1)
+                .orderBy()
+                .desc("scheduled")
+                .findOneOrEmpty()
+                .map(DatabaseExecutionHelper::toInternalModel);
```

Comment:
I don't see this class in the review, so I assume it's an existing class. Generally, not a fan of static helper classes, and while we do have them in places, it's generally better to create instances of them in order to facilitate testing and injection. If this is indeed existing code, feel free just to add a `TODO` into that class.